src/or/proto_socks.c

```diff
@@ -287,6 +403,45 @@ fetch_from_buf_socks(buf_t *buf, socks_request_t *req,
     buf_clear(buf);
     res = 1;
     goto end;
+  } else if (socks_version == 5) {
+    if (datalen < 2) { /* version and another byte */
+      res = 0;
+      goto end;
+    }
+
+    if (!req->got_auth) {
+
+    }
+
+    if (req->socks_version != 5) {
+      int have_user_pass, have_no_auth;
```

Comment:
Shouldn't these be initialized to 0, given how they aren't set above?  I would suggest doing _both_.  (Initialize to 0, and set to 0 in parse function)

Reply:
I think I did that in some later commit.